core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -18,280 +18,416 @@
+    this.focusManagerService.setFocus('tmpFocusPoint');
+    for (let i = 0; i < this.customizationArgSpecs.length; i++) {
+      const caName = this.customizationArgSpecs[i].name;
       if (caName === 'math_content') {
-        $scope.currentRteIsMathExpressionEditor = true;
-        var mathValueDict = {
+        this.currentRteIsMathExpressionEditor = true;
+        const mathValueDict = {
           name: caName,
-          value: (
-            attrsCustomizationArgsDict.hasOwnProperty(caName) ?
-              angular.copy(attrsCustomizationArgsDict[caName]) :
-              customizationArgSpecs[i].default_value)
-        };
+          value: this.attrsCustomizationArgsDict.hasOwnProperty(caName) ?
+            typedCloneDeep(this.attrsCustomizationArgsDict[caName]) :
+            this.customizationArgSpecs[i].default_value,
+        } as Extract<
+          CustomizationArgsNameAndValueArray[number],
+          { name: typeof caName }
+        >;
         // If the component being created or edited is math rich text component,
         // we need to pass this extra attribute svgFile to the math RTE editor.
         // The math RTE editor will auto-generate the svgFile based on the
         // rawLatex value and then this file can be saved to the backend when
         // the user clicks on the save button.
         mathValueDict.value.svgFile = null;
         mathValueDict.value.mathExpressionSvgIsBeingProcessed = false;
-        $scope.tmpCustomizationArgs.push(mathValueDict);
+        (
+          this.tmpCustomizationArgs as Extract<
+            CustomizationArgsNameAndValueArray[number],
+            { name: typeof caName }
+          >[]
+        ).push(mathValueDict);
       } else {
-        $scope.tmpCustomizationArgs.push({
+        const t = {
           name: caName,
-          value: (
-            attrsCustomizationArgsDict.hasOwnProperty(caName) ?
-              angular.copy(attrsCustomizationArgsDict[caName]) :
-              customizationArgSpecs[i].default_value)
-        });
+          value: this.attrsCustomizationArgsDict.hasOwnProperty(caName) ?
+            angular.copy(this.attrsCustomizationArgsDict[caName]) :
+            this.customizationArgSpecs[i].default_value,
+        } as Extract<
+          CustomizationArgsNameAndValueArray[number],
+          { name: typeof caName }
+        >;
+        (
+          this.tmpCustomizationArgs as Extract<
+            CustomizationArgsNameAndValueArray[number],
+            { name: typeof caName }
+          >[]
+        ).push(t);
       }
     }
     // Infer that the RTE component is a Link if it contains the `url` and
     // `text` customization arg names.
-    const customizationArgNames = customizationArgSpecs.map(x => x.name);
-    if (customizationArgNames.includes('url') &&
-        customizationArgNames.includes('text')) {
-      $scope.currentRteIsLinkEditor = true;
+    const customizationArgNames = (
+      this.customizationArgSpecs as { name: string }[]
```

Comment:
Why do we need to redefine the type here?

Reply:
Thanks, see above for what to do here (comment + TODO).

Reply:
Done